scripts/biomedical/Human-GEM_Human1/scripts/format_metabolites.py

```diff
@@ -0,0 +1,466 @@
+    with data commons
+    Args:
+        kegg_dcids: list of kegg dcids
+    Returns:
+        data common query result with chembl\
+        and kegg dcid mapping
+    """
+    query_str = """
+    SELECT DISTINCT ?chembl ?kegg
+    WHERE{{
+    ?chembl typeOf ChemicalCompound .
+    ?chembl keggCompoundID ?kegg .
+    ?chembl keggCompoundID {0} .
+    }}
+    """.format(kegg_dcids)
+    result = dc.query(query_str)
+    return result
+
+
+def query_chembl_from_hmdb(hmdb_dcids):
+    """query chembl id from hmdb id
+    with data commons
+    Args:
+        hmdb_dcids: list of hmdb dcids
+    Returns:
+        data common query result with chembl\
+        and hmdb dcid mapping
+    """
+    query_str = """
+    SELECT DISTINCT ?chembl ?hmdb
+    WHERE{{
+    ?chembl typeOf ChemicalCompound .
+    ?chembl humanMetabolomeDatabaseID ?hmdb .
+    ?chembl humanMetabolomeDatabaseID {0} .
+    }}
+    """.format(hmdb_dcids)
+    result = dc.query(query_str)
+    return result
+
+
+def make_csv(df, name):
+    """create csv file at the
+    current directory path
+    Args:
+        df: pandas dataframe to be created
+        name: name of csv file (e.g result.csv)
+    Returns:
+        None
+    """
+    output_path = os.path.join(os.getcwd(), name)
+    df.to_csv(output_path, index=None)
+    return None
+
+
+def convert_float_to_int(df, column):
+    """convert type of pd.Datafarme column
+    from float to int, ignore missing values
+    Args:
+        df: pandas.DataFrame
+        column: column name
+    Returns:
+        None
+    """
+    df[column] = df[column].fillna(-1)
+    df[column] = df[column].astype(int)
+    df[column] = df[column].astype(str)
+    df[column] = df[column].replace('-1', np.nan)
+    return None
+
+def format_product_role_data(product_tsv, df_metabolic_cellular_compartment):
+    """Format product role data
+    Args:
+        product_tsv: productRoles.tsv file path
+        df_metabolic_cellular_compartment:
+    Returns:
+        df_product_roles: formattted product role dataframe
+    """
+    ### generate productRoles.csv
+    df_product_roles = pd.read_csv(product_tsv, sep='\t')
+    # Remove "M_" in speciesID/humanGEMID of metabolites
+    df_product_roles["speciesID"] = df_product_roles["speciesID"].str[2:]
+    # Remove "R_" in speciesID/humanGEMID of reactions
+    df_product_roles["reactionID"] = df_product_roles["reactionID"].str[2:]
+    # merge productRoles and metabolicCellularCompartment for dcid mapping
+    df_product_roles = df_product_roles.merge(\
+                df_metabolic_cellular_compartment,\
+                left_on="speciesID", right_on="id")\
+                [["reactionID", "metabolic_compartment_dcid"]]
+    # modify reaction id to reaction dcid format
+    df_product_roles["reactionID"] = "bio/" +\
+                  df_product_roles["reactionID"].astype("str")
+    return df_product_roles
+
+def format_reactant_role_data(reactant_tsv, df_metabolic_cellular_compartment):
```

Comment:
The code in this method is pretty much the same as format_product_role_data. Can actually just be one method.